hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/util/Caches.java

```diff
@@ -291,6 +291,10 @@ public static void removeAll(AdvancedCache cache) {
 	}
 
 	public static CollectableCloseableIterable keys(AdvancedCache cache) {
+		if (cache.getCacheConfiguration().transaction().transactionMode().isTransactional()) {
+			// Dummy read to enlist the LocalTransaction as workaround for ISPN-5676
+			cache.containsKey(false);
```

Comment:
Also this will be fixed in ISPN 7.2.5 which will render this change unneeded.  But I am not for sure when that will be released.